services/ux-backend/handlers/rotatekeys/handler.go

```diff
@@ -0,0 +1,90 @@
+package rotatekeys
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	contentTypeTextPlain                     = "text/plain"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+	onboardingValidationPrivateKeySecretName = "onboarding-private-key"
+)
+
+var (
+	namespace string
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client) {
+	switch r.Method {
+	case "POST":
+		handlePost(r.Context(), w, cl)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(ctx context.Context, w http.ResponseWriter, cl client.Client) {
+	klog.Info("POST method on /rotate-keys endpoint is invoked")
+	w.Header().Set("Content-Type", contentTypeTextPlain)
+
+	if namespace == "" {
+		var err error
+		namespace, err = getNamespace()
+		if err != nil {
+			klog.Errorf("failed to get current namespace: %v", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			if _, err = w.Write([]byte("Failed to rotate keys")); err != nil {
+				klog.Errorf("failed to write date to response writer, %v", err)
+			}
+			return
+		}
+	}
+
+	publicKeySecret := &corev1.Secret{}
+	publicKeySecret.Name = onboardingValidationPublicKeySecretName
+	publicKeySecret.Namespace = namespace
+	err := cl.Delete(ctx, publicKeySecret)
+	if err != nil {
+		klog.Errorf("failed to delete public key secret: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+
+		// TODO: should we differentiate b/n secret not found and remaining errors?
+		if _, err = w.Write([]byte("Failed to rotate keys")); err != nil {
+			klog.Errorf("failed to write date to response writer, %v", err)
+		}
+		return
+	}
+
+	klog.Info("onboarding keys are rotated successfully")
+	w.WriteHeader(http.StatusOK)
+	if _, err = w.Write([]byte("Successfully rotated keys")); err != nil {
+		klog.Errorf("failed to write date to response writer, %v", err)
+	}
+}
+
+func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
+	klog.Info("Only POST method should be used to send data to this endpoint /rotate-keys")
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	w.Header().Set("Content-Type", contentTypeTextPlain)
+	w.Header().Set("Allow", "POST")
+
+	if _, err := w.Write([]byte(fmt.Sprintf("Unsupported method : %s", r.Method))); err != nil {
+		klog.Errorf("failed to write data to response writer: %v", err)
+	}
+}
+
+// getNamespace returns current namespace that the pod is running
+func getNamespace() (string, error) {
+	f, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
Why are we reading from a file? what was wrong with reading from an envvar? The request from the prev review was to not include this in the server config struct, not to move to read from a file.

Reply:
done.